src/adapters/suggest.js

```diff
@@ -1,97 +1,116 @@
 const geocoderConfig = nconf.get().services.geocoder;
 const SUGGEST_MAX_ITEMS = geocoderConfig.maxItems;
 const SUGGEST_USE_FOCUS = geocoderConfig.useFocus;
 const SUGGEST_FOCUS_MIN_ZOOM = 11;
 
-import { suggestResults } from './suggest_sources';
-import SuggestsDropdown from '../components/ui/SuggestsDropdown';
 
 export default class Suggest {
-  constructor({ tagSelector, onSelect,
-    withGeoloc = false, withCategories = false,
+
+  constructor({
+    tagSelector,
+    onSelect,
+    withGeoloc = false,
+    withCategories = false,
   }) {
-    this.searchInputDomHandler = document.querySelector(tagSelector);
-    this.poi = null;
-    this.suggestList = [];
+
+    this.tagSelector = tagSelector;
     this.onSelect = onSelect;
+    this.withGeoloc = withGeoloc;
+    this.withCategories = withCategories;
+
+    this.searchInputDomHandler = document.querySelector(tagSelector);
 
     this.autocomplete = new Autocomplete({
       selector: tagSelector,
       minChars: 0,
       delay: 100,
-      source: term => suggestResults(term, {
-        withCategories,
-        useFocus: SUGGEST_USE_FOCUS,
-        focusMinZoom: SUGGEST_FOCUS_MIN_ZOOM,
-        maxFavorites: !term ? 5 : 2,
-        maxItems: SUGGEST_MAX_ITEMS,
-      }),
-      renderItems: items => {
-        const firstFav = items.findIndex(item => item instanceof PoiStore);
-        if (firstFav !== -1) {
-          items.splice(firstFav, 0, { simpleLabel: _('Favorites', 'autocomplete').toUpperCase() });
-        }
-        if (withGeoloc) {
-          items.splice(0, 0, NavigatorGeolocalisationPoi.getInstance());
-        }
-
-        // Create a react node, or reuse the existing node
-        const existingElem = document.getElementById('react-suggests-' + tagSelector);
-        let elem = null;
-        if (!existingElem) {
-          elem = document.createElement('div');
-          elem.setAttribute('id', 'react-suggests-' + tagSelector);
-          this.searchInputDomHandler.parentNode.appendChild(elem);
-        }
-
-        const reactElem = existingElem || elem;
-        const typedValue = this.searchInputDomHandler.value;
-
-        ReactDOM.render(
-          <SuggestsDropdown
-            inputId={this.searchInputDomHandler.getAttribute('id')}
-            suggestItems={items}
-            onHighlight={item => {
-              this.searchInputDomHandler.value = item && item.name || typedValue;
-            }}
-            onSelect={item => {
-              this.searchInputDomHandler.value = item.name || '';
-              this.searchInputDomHandler.blur();
-              this.onSelect(item);
-            }}
-          />
-          , reactElem
-        );
-      },
+      source: term => this.suggestResults(term),
+      renderItems: items => this.renderItems(items),
     });
 
-    this.searchInputDomHandler.addEventListener('blur', function handleSearchInputBlur() {
-      unmountReactSuggestDropdown();
+    this.searchInputDomHandler.addEventListener('blur', () => {
```

Comment:
I just realized we never removed this listener… So we probably have always had memleaks when we open/close the direction panel, as we reinstantiate suggests multiple times. Hopefully the next refacto steps will make it easier to solve that.